src/test/java/com/kodilla/ecommercee/repositoryDao/CartRepositoryTestSuit.java

```diff
@@ -0,0 +1,200 @@
+        listOfProducts2 = new ArrayList<>();
+        listOfProducts3 = new ArrayList<>();
+
+        cart1 = new Cart();
+        cart1.setProductsList(listOfProducts1);
+        cart2 = new Cart();
+        cart2.setProductsList(listOfProducts2);
+        cart3 = new Cart();
+        cart3.setProductsList(listOfProducts3);
+    }
+    @Transactional
+    @Test
+    public void testCartDaoSaveAndFindAll(){
+        //Given
+        cartRepository.save(cart1);
+        cartRepository.save(cart2);
+        cartRepository.save(cart3);
+        //When
+        List<Cart> listOfCarts = cartRepository.findAll();
+        //Then
+        Assert.assertEquals(3,listOfCarts.size());
+        Assert.assertNotEquals(null,cartRepository.findById(cart1.getId()));
+        Assert.assertNotEquals(null,cartRepository.findById(cart2.getId()));
+        Assert.assertNotEquals(null,cartRepository.findById(cart3.getId()));
+        //Clean Up
+        try{
+            cartRepository.deleteById(cart1.getId());
+            cartRepository.deleteById(cart2.getId());
+            cartRepository.deleteById(cart3.getId());
+        }catch (Exception e){
+            LOGGER.error("Clean-up process failed.", e.getMessage(), e);
+        }
+    }
+    @Transactional
+    @Test
+    public void testCartDaoSaveAndFindById(){
+        //Given
+        cartRepository.save(cart1);
+        cartRepository.save(cart2);
+        cartRepository.save(cart3);
+        Long IdCart1 = cart1.getId();
+        //When
+        Optional<Cart> cartFound = cartRepository.findById(IdCart1);
+        //Then
+        Assert.assertEquals(IdCart1,cartFound.get().getId());
+        //Clean Up
+        try{
+            cartRepository.deleteById(cart1.getId());
+            cartRepository.deleteById(cart2.getId());
+            cartRepository.deleteById(cart3.getId());
+        }catch (Exception e){
+            LOGGER.error("Clean-up process failed.", e.getMessage(), e);
+        }
+    }
+    @Transactional
+    @Test
+    public void testCartDaoDeleteById(){
+        //Given
+        cartRepository.save(cart1);
+        cartRepository.save(cart2);
+        cartRepository.save(cart3);
+        Long cart2Id = cart2.getId();
+        //When
+        cartRepository.deleteById(cart2Id);
+        List<Cart> cartList = cartRepository.findAll();
+        //Then
+        Assert.assertFalse(cartList.contains(cart2));
+        //Clean Up
+        try{
+            cartRepository.deleteById(cart1.getId());
+            cartRepository.deleteById(cart3.getId());
+        }catch (Exception e){
+            LOGGER.error("Clean-up process failed.", e.getMessage(), e);
+        }
+    }
+
+    @Transactional
+    @Test
+    public void testCartOneOnOneWithOrder(){
+        //Given
+        Order order = new Order();
+        cart1.setOrder(order);
+        //When
+        cartRepository.save(cart1);
+        Assert.assertNotEquals(zeroLong,cartRepository.findById(cart1.getOrder().getOrderId()));
```

Comment:
findById -> Optional

Reply:
nie wiem o co Ci chodzi ze w tej mojej asserci wywoluje Optionala, to jak mam to rozwiazac?

Reply:
Ten Assert:  Assert.assertNotEquals(zeroLong,cartRepository.findById(cart1.getOrder().getOrderId()));
porwnuje Long Vs Optional czyli zawsze wybik bedzie NotEquals - to nie ma sensu.
Musisz sprawdzic czy jest wartosc w Optional i pobrac z niej id.  Nastepnie porownujesz wartosc tego Id z zeroLong.

Reply:
Optional<Cart> orderId = cartRepository.findById(cart1.getOrder().getOrderId());
        //Then
        Assert.assertNotEquals(zeroLong,orderId);

@erwinMuz czy ten zapis bedzie poprawny o to chodzilo?